### PR TITLE
Do not cause control client status call to fail on canceled

### PR DESCRIPTION
**Pull request by vladaionescu**

### CODE REVIEW

File: client/solve.go [hunk 1]
```diff
@@ -326,6 +328,9 @@ func (c *Client) solve(ctx context.Context, def *llb.Definition, runGateway runG
 				if err == io.EOF {
 					return nil
 				}
+				if status.Code(errors.Cause(err)) == codes.Canceled {
```
Comment (alexcb) [inline client/solve.go:332]: can you add a comment such as `// earthly-specific: ...` to make it more clear this was added by us. This makes it much easier to deal with upstream merges.

**Review by alexcb — COMMENTED** [review]
I think it's fine.

Can you link to the corresponding earthly PR that references this branch to show that tests continue to work? It's documented under https://github.com/earthly/earthly/blob/main/CONTRIBUTING.md#updates-to-buildkit-or-fsutil

Once they are working, you can then merge this PR via a fast-forward merge using git directly, which will merge this without changing the commit sha (which means the corresponding earthly PR doesn't have to get updated to reflect a new commit sha).

File: client/solve.go (hunk 1 above)
Comment (vladaionescu) [reply client/solve.go:332]: Done

**Comment by vladaionescu** [comment]
Here we go... https://github.com/earthly/earthly/pull/3204

**Review by alexcb — APPROVED** [review]

**Pull request merged**